File: app/src/main/java/com/gerigorila/agenticmigrationdemo/presentation/productdetail/ProductDetailContract.java
package com.gerigorila.agenticmigrationdemo.presentation.productdetail;

import com.gerigorila.agenticmigrationdemo.domain.model.Product;

public interface ProductDetailContract {

    interface View {
        void showProduct(Product product);
        void showError(String message);
    }

    interface Presenter {
        void loadProduct(int productId);
    }
}

File: app/src/main/java/com/gerigorila/agenticmigrationdemo/presentation/productdetail/ProductDetailContract.kt
package com.gerigorila.agenticmigrationdemo.presentation.productdetail

import com.gerigorila.agenticmigrationdemo.domain.model.Product

interface ProductDetailContract {

    interface View {
        fun showProduct(product: Product)
        fun showError(message: String)
    }

    interface Presenter {
        fun loadProduct(productId: Int)
    }
}
